packages/falso/src/lib/core/unique-validators.ts

```diff
@@ -0,0 +1,32 @@
+export const primitiveValueIsUnique: <T>(item: T, items: T[]) => boolean = (
+  item,
+  items
+) => !items.includes(item);
+
+export const dateIsUnique: (date: Date, dates: Date[]) => boolean = (
+  date,
+  dates
+) => !dates.some((d) => d.valueOf() === date.valueOf());
+
+export const objectWithIdIsUnique: <T extends { id: string }>(
+  item: T,
+  items: T[]
+) => boolean = (item, items) => objectIsUnique(item, items, ['id']);
+
+export const objectIsUnique: (
+  item: any,
+  items: any[],
+  keys: string[]
+) => boolean = (item: any, items: any[], keys: string[]) => {
+  for (const key of keys) {
+    if (!item[key]) {
+      throw new Error(`${key} does not exist in this array value type`);
+    }
```

Comment:
I think we can protect this using types no?

Reply:
I'm not sure what you mean. I have made the function a little more type safe by removing the any. I've left the error in for devs as it will catch any typos in the keys they pass.

```ts
export function objectIsUnique<T extends Record<string, any>>(
  item: T,
  items: T[],
  keys: string[],
): boolean {
  for (const key of keys) {
    if (!(key in item)) {
      throw new Error(`${key} does not exist in this array value type`);
    }

    if (items.some((arrayItem) => arrayItem[key] === item[key])) {
      return false;
    }
  }

  return true;
}
```

Reply:
Sorry, the coin has just dropped with what you meant. The function now reads as:

```ts
export function objectIsUnique<T extends Record<string, any>>(
  item: T,
  items: T[],
  keys: (keyof T)[]
): boolean {
  for (const key of keys) {
    if (items.some((arrayItem) => arrayItem[key] === item[key])) {
      return false;
    }
  }

  return true;
}
```